### DISCUSSION TITLE
Use of StringEnumConverter without attribute

### DISCUSSION THREAD

**Discussion by Ghostic** [VerifyTests/Verify#602] [Q&A]
Hello I'm trying to introduce this amazing package to our team but I have a problem with enums.

The class that I'm verifying is auto generated so I can't use any attributes.

I have tried creating custom resolvers like:
```
class CustomResolver : DefaultContractResolver
        {
            protected override JsonObjectContract CreateObjectContract(Type objectType)
            {
                JsonObjectContract contract = base.CreateObjectContract(objectType);
                if (objectType == typeof(Enum))
                {
                    contract.Converter = new StringEnumConverter();
                }
                return contract;
            }
        }
```
and adding it to settings:

```
settings.AddExtraSettings(x => x.ContractResolver = new CustomResolver());
```

I have also tried adding it to converts like:
```
settings.AddExtraSettings(_ => _.Converters.Add(new StringEnumConverter()));
```
but I don't think it's needed as I have seen in a code that is already using that converter. I just struggle with telling Verify to apply that conversion without using any attributes. 

Any ideas?

**Comment by SimonCropp** [comment]
can u share the generated code of the class?

**Reply by Ghostic** [reply]
Hello thank you for a response.

This class is generated with visual studio using WSDL contract. Here are examples of some Enums:
```
[System.Xml.Serialization.XmlAttributeAttribute()]
public EnumType EnumName
        {
            get
            {
                return this.backingField;
            }
            set
            {
                this.backingField= value;
            }
        }
```
and some of them are decorated with two attributes:

```
[System.Xml.Serialization.XmlAttributeAttribute()]
[System.ComponentModel.DefaultValueAttribute(EnumType)]
public EnumType EnumName
        {
            get
            {
                return this.backingField;
            }
            set
            {
                this.backingField= value;
            }
        }
```

They are in a base object of the class I'm serializing and some other types like int or string marked with:
```
[System.Xml.Serialization.XmlAttributeAttribute()]
```
work fine as I'm getting them in a generated txt file.

--------------------------------------------------------------------------------------
Edit

I have just noticed that definitions of these enum are marked with 
```
[System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.Tools.ServiceModel.Svcutil", "2.0.3")]
 [System.Xml.Serialization.XmlTypeAttribute(Namespace="some namespace here")]
```

**Reply by SimonCropp** [reply]
so EnumType  is really an enum? since enums should already be serialized as strings

**Reply by Ghostic** [reply]
Yes it is normal enum like:
```
/// <remarks/>
    [System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.Tools.ServiceModel.Svcutil", "2.0.3")]
    [System.Xml.Serialization.XmlTypeAttribute(Namespace="some name space")]
    public enum Option
    {
        
        /// <remarks/>
        Option1,
        
        /// <remarks/>
        Option2,
        
        /// <remarks/>
        Option3,
    }
```

**Reply by SimonCropp** [reply]
can u submit a failing unit test as a pull request

**Reply by Ghostic** [reply]
I had problems building your solution on my PC. But I have created really simple test to reproduce the issue.
https://github.com/Ghostic/EnumSample
